cryptographic_flows.md

```diff
@@ -22,8 +22,36 @@ An asset owner that has not previously interacted with the key server MUST regis
     1. Checks that `user_id` matches that of the authenticated user in the open registration session. If this check fails, the server MUST abort.
     1. Checks that the received ciphertext is of the expected format and length.
     1. [Stores](#server-side-storage) the received ciphertext in a record associated with `user_id`.
+    1. Stores the registration registration request in the [audit log](#audit-logs).
 1. The client and server close the session upon completion of the request.
 
+## Retrieve Audit Logs
+The asset owner can request audit logs from the key server.
+
+Input:
+- `user_id`, a 128-bit globally unique identifier (GUID) representing the identity of the asset owner.
+- `type`, one of:
+    - `"system only"`, which indicates the asset owner wants a record of registration and login actions;
+    - `"key only"`, which indicates the asset owner wants a record of requested key use operations with respect to or more keys; or
+    - `"all"`, which indicates the asset owner wants both system and requested key use operations.
+- `key_identifiers`, an OPTIONAL list of key identifiers.
```

Comment:
If there is no `key identifier` provided but the `type` is `“key only”` is the output the logs for all the keys?

Reply:
That would make sense. It should probably do the same for "all" with no key identifier provided, right?

Reply:
Yes I think that makes sense. Added a bit about that in the `key_identifiers` description.